web/js/playground.js

```diff
@@ -101,33 +95,15 @@ function renderTools() {
     toolSetContainer.innerHTML = '';
 
     const recentTools = getRecentlyUsedTools();
-    const hasRecentlyUsedTools = recentTools.length > 0;
-
-    // Divide ToolsCatalog into recently used tools and non-recently-used tools
-    const recentlyUsedLabels = recentTools.map(tool => tool.label.toLowerCase());
-    const recentToolsSet = new Set(recentlyUsedLabels);
-
-    const recentlyUsedTools = ToolsCatalog.filter(tool =>
-        recentToolsSet.has(tool.label.toLowerCase())
-    );
+    //const recentToolLabels = new Set(recentTools.map(tool => tool.label.toLowerCase()));
 
-    const nonRecentlyUsedTools = ToolsCatalog.filter(tool =>
-        !recentToolsSet.has(tool.label.toLowerCase())
-    );
-
-    // Sort recently used tools based on their order in recentTools array
-    const sortedRecentlyUsedTools = recentlyUsedTools.sort(
-        (a, b) =>
-            recentToolsSet.has(a.label.toLowerCase()) - recentToolsSet.has(b.label.toLowerCase())
-    );
-
-    // Combine sorted recently used tools and non-recently-used tools
-    const sortedTools = [...sortedRecentlyUsedTools, ...nonRecentlyUsedTools];
+    const sortedTools = [...recentTools, ...ToolsCatalog.filter(tool => 
+        !recentTools.some(recentTool => recentTool.label === tool.label))];
```

Comment:
I can only find where the list of recent tools is passed between functions within the code, and not where it is initially defined. What exactly moves a tool to the top of the list when it is used?